target/arm/helper.c

```diff
@@ -11755,11 +11755,14 @@ static bool get_phys_addr_lpae(CPUARMState *env, uint64_t address,
 
     int lc = extract32(hwu, 2, 2);
     int sc = extract32(hwu, 1, 1);
-    int cdbm = extract32(hwu, 0, 1);
 
-    // Cap stores can fault here as only tagged stores specify
-    // MMU_DATA_CAP_STORE
-    if (!cdbm && !sc) {
+    /*
+     * Cap stores can fault here as only tagged stores specify
+     * MMU_DATA_CAP_STORE.  Unlike the hardware, we raise traps on SC being
```

Comment:
This doesn't read quite right to me. The hardware does also raise traps on SC being clear, in the case that CDBM is also clear. I'd suggest:

> Cap stores can fault here as only tagged stores specify MMU_DATA_CAP_STORE. QEMU lacks FEAT_HAFDBS support so we trap based on SC alone.

Note also there is a recent patch series floating around to add FEAT_HAFDBS to QEMU